### ISSUE TITLE
Simplify Javascript Object Notation Patch

### ISSUE THREAD

**Issue by yinho999** [loco-rs/loco#452]
## Feature Request

Based on my understanding, users often find it challenging to implement [JavaScript Object Notation Patch (JSON Patch)](https://datatracker.ietf.org/doc/html/rfc6902) as outlined in RFC 6902 and RFC7396. 

To facilitate this, we could integrate a [crate](https://crates.io/crates/json-patch) specifically for handling UpdateParams in views, making it an optional feature. This approach ensures flexibility, as not all users may require JSON Patch functionality.

**Comment by jondot** [comment]
How would people use this? in controller?
can you show a few examples maybe?

**Comment by yinho999** [comment]
> How would people use this? in controller? can you show a few examples maybe?

Yeah I think we can put that into DTO and I will provide some examples as soon as I am free 👍️

**Comment by yinho999** [comment]
@jondot, here is a simple (example)[https://github.com/yinho999/loco-playground/blob/json-patch/src/bin/json-patch.rs] of using the json patch crate.

However, as you can see, the Patch type is very generic and does not offer any type safety or restrictions. For instance, JSON Patch has an operation called "replace" that can be applied to any field of the user to change its value. This is what I use to update the user's email in the example. However, we can also apply different operations like "test", "remove", or "add" on the same field.

I suggest we could add operation-based functions similar to what Spring Boot did, to restrict operations and provide a simple API for users [Spring Boot](https://github.com/spring-projects/spring-data-rest/blob/f92169d7fb66026bbb6615bcbc30375002f171d1/spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/json/patch/PatchOperation.java#L146). Moreover we can also provide macros to serialize the Patch JSON into multiple operations DTOs to provide type safety.

**Comment by kaplanelad** [comment]
Hey, @yinho999!

Based on the simple example, it looks like you just need to `include json_patch::Patch` and update the controller extractor. This change can be seamlessly used by users without integrating it into the Loco infrastructure.

I'm closing this for now. If you find it still relevant or have other ideas, feel free to reopen this issue.
